joomla/joomla-cms
# [4.0] Flags are country flags, not language flags

During the Joomla installation, there is an option to automatically create a multilingual site with the content languages already created. That is awesome.

However the logic for the language flags applied there is no proper logic at all: The issue is that there exists no language flags, they are country (or district) flags.

When a new official language is created by our translation teams, we manually have to upload a new image matching the new language code. For the german dialects this was done recently in a PR (see https://github.com/joomla/joomla-cms/pull/11868).
### Summary of Changes

This PR suggest to change the logic and assign the flags based on the country code in our language codes. Eg for de-CH (Swiss German) it would take the image `ch.gif` instead of `de-ch.gif`.
To be backward compatible it will first look for an existing file with the language code and fall back to the country code.

This PR also removes all unneeded files and renames some:
- 33 files are renamed to their corresponding country code.
- 72 files are deleted because they were duplicates and are no longer needed.

The deleted and renamed files are put into a README file so we don't add files with the same name in future (and thus overwrite existing ones on user sites).
### Known issue

The limitation here is that the languages for multilingual countries like Switzerland (in theory de-CH, fr-CH, it-CH and rm-CH) would share the same flag. The alternative is to create ugly, arbitary flags like https://github.com/joomla/joomla-cms/blob/staging/media/mod_languages/images/de_ch.gif so you get different flags for each language.
~~We would also have to update the images we ship to make sure the folder contains all country flags. Currently, it is a mix between flags named after a language (eg `en-gif` for en-GB) and country (eg `us.gif` or `ch.gif`), plus a flag for each language code (`en-GB.gif` and `en-US.gif`). So we have a lot of duplicate and inconsistently named files here (see https://github.com/joomla/joomla-cms/tree/staging/media/mod_languages/images).~~ Already done now.
### Testing Instructions
- Install Joomla and select to install some languages. Activate the option to create a multilanguage site and to create localised content.
- Check that the created content languages have proper flags for the country they are referring.
### Documentation Changes Required

None that I am aware of.
